src/Test.UnitTests.Sarif.Converters/TestData/FortifyFprConverter/ExpectedOutputs/TwoResultsWithNodeRefs.sarif

```diff
@@ -391,13 +266,153 @@
         {
           "location": {
             "uri": "Program.cs",
-            "uriBaseId": "SRCROOT",
-            "index": 2
+            "uriBaseId": "SRCROOT"
           },
           "length": 5009,
           "mimeType": "text/x-csharp",
           "encoding": "windows-1252"
         }
+      ],
+      "threadFlowLocations": [
+        {
+          "location": {
+            "physicalLocation": {
+              "artifactLocation": {
+                "index": 2
+              },
+              "region": {
+                "startLine": 82,
+                "snippet": {
+                  "text": "                            Logger.Error(ex.ToString());"
+                }
+              },
+              "contextRegion": {
+                "startLine": 79,
+                "endLine": 85,
+                "snippet": {
+                  "text": "                                Logger.Error(LogMessage.RETRY_ATTEMPTS_EXCEEDED, new object[] { tool, server});\n                                continue;\n                            }\n                            Logger.Error(ex.ToString());\n                        }\n                    }\n\n"
+                }
+              }
+            },
+            "message": {
+              "text": "Error(0)"
```

Comment:
>"text": "Error(0)" [](start = 14, length = 18)

Is this really the message? Seems like there's something further to resolve here. Probably not a gap (if one) that you introduced. open issue for it?

Reply:
This comes from the <Action> in the <Node> in the stack. It looks like for this rule they are the function name and argument, but it varies. It looks like this is a fallback path for when there isn't an explicit "NodeLabel" in the FVDL to use instead.

It's from here in the FVDL we converted:
`
<UnifiedNodePool>
  <Node id="4">
    <SourceLocation path="Program.cs" line="82" lineEnd="82" colStart="0" colEnd="0" contextId="2" snippet="F64E01C1BF6395D7221768BE7CC7BBB9#Program.cs:82:82"/>
    <Action type="InCall">Error(0)</Action>
    <Knowledge>
      <Fact primary="false" type="Call">Direct : Helpers.Logger.Error</Fact>
    </Knowledge>
  </Node>
`